uktrade/data-hub-frontend
# Add Postcode Validation to postcode field when country is United States or Canada

## Description of change

This PR adds front-end validation to the add/edit company form to mark Postcode as a mandatory field when the country is United States or Canada.
Also corrects the Post Code terminology to match the regional equivalents (ZIP code for US and postal code for Canada)

## Test instructions

Go to add a new company/ edit an existing company that has a country set as United States or Canada.
If you attempt to submit the form without completing the postcode field, the field should be highlighted with an error message.

## Screenshots
### Before

![image](https://user-images.githubusercontent.com/20663545/129217768-dc7a47ae-f158-4d5c-9590-7f9967da62f2.png)

### After
![image](https://user-images.githubusercontent.com/70902973/129367192-ca316008-85e9-413f-a293-e75f181fbbd5.png)
![image](https://user-images.githubusercontent.com/70902973/129366854-a4fe3fa8-6910-45b1-824d-10e1999aee98.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
